src/View/MvcMailViewRenderer.php

```diff
@@ -28,12 +29,25 @@ public function render(string $name, array $params = []): string
 
         // If a layout was provided, add the original view model as a child
         if ($layout !== null) {
-            $layoutModel = new ViewModel([
-                'content' => $this->renderer->render($viewModel),
-            ]);
-            $layoutModel->setTemplate($layout);
+				            
+            $childTemplateName = $params['child_template_name'] ?? 'content';
+            $layoutModelParams = [
+                $childTemplateName => $this->renderer->render($viewModel)
+            ];
+				            
+            if ( isset($params['layout_params']) ) {
+                if ( is_bool($params['layout_params'] && $params['layout_params'] === true) ) {
+                   $layoutModelParams = array_merge($layoutModelParams, $params);
+                }
+                if ( is_array($params['layout_params']) ) {
+                    $layoutModelParams = array_merge($layoutModelParams, $params['layout_params']);
+	        }
+	    }
+	
+	    $layoutModel = new ViewModel($layoutModelParams);
+	    $layoutModel->setTemplate($layout);
             $viewModel = $layoutModel;
```

Comment:
I think this approach might be a little bit overengineered. We don't need to make all of those checks, but just always inject the params both in the layout and the inner template. If the user wants to access them, they will be there as expected, if not, the fact that they have been actually injected won't have any side effect.

So, I would try with something like this:

```php
if ($layout !== null) {
    $childTemplateName = $params['child_template_name'] ?? 'content';
    $params[$childTemplateName] = $this->renderer->render($viewModel);

    $viewModel = new ViewModel($params);
    $viewModel->setTemplate($layout);
}
```

I have not tested it, but I think that using this simple approach the problem should be solved.